### PR TITLE
Add semantics for if-then-else conditionals

### PR DESCRIPTION
**Pull request by philectron**
Add this `if-then-else` conditional to satisfy the 50% feature requirement.

Manual test results (will use these examples later in the [`tests` branch](https://github.com/philectron/lil-lisp/tree/tests):

image

### CODE REVIEW

File: Main.hs [hunk 1]
```diff
@@ -14,7 +14,7 @@ expr (N v) = N v
 expr (S v) = S v
 expr (List v) = List v
 expr (Error v) = S v
-expr (If cond exprl exprr) = ifExpr cond exprl exprr
+expr (If cond exprThen exprElse) = ifExpr cond exprThen exprElse
```
Comment (khashf) [inline Main.hs:17]: What about the case when the `else` expression is optional?

Comment (philectron) [reply Main.hs:17]: image

Comment (philectron) [reply Main.hs:17]: @devsebb said earlier in the planning phase that we can just pass `Nil` into the else clause.

Comment (philectron) [reply Main.hs:17]: We'll need to add `Nil` to the `Expr` grammar in order to use it, though.

Comment (khashf) [reply Main.hs:17]: Let's require always have an `else` expression. no non-else if expression. then, if user doesn't want to do anything, just put empty ()

**Review by khashf — APPROVED** [review]

**Pull request merged**